server/mailet/integration-testing/src/test/java/org/apache/james/mailets/MimeMessageBenchTest.java

```diff
@@ -0,0 +1,146 @@
+import static org.apache.james.mailets.configuration.Constants.LOCALHOST_IP;
+import static org.apache.james.mailets.configuration.Constants.PASSWORD;
+import static org.apache.james.mailets.configuration.Constants.awaitAtMostOneMinute;
+
+import java.util.concurrent.TimeUnit;
+import java.util.stream.IntStream;
+
+import org.apache.james.MemoryJamesServerMain;
+import org.apache.james.core.builder.MimeMessageBuilder;
+import org.apache.james.mailets.configuration.CommonProcessors;
+import org.apache.james.mailets.configuration.MailetConfiguration;
+import org.apache.james.mailets.configuration.MailetContainer;
+import org.apache.james.mailets.configuration.ProcessorConfiguration;
+import org.apache.james.modules.protocols.SmtpGuiceProbe;
+import org.apache.james.probe.DataProbe;
+import org.apache.james.server.core.MailImpl;
+import org.apache.james.transport.mailets.LogMessage;
+import org.apache.james.transport.matchers.RecipientIs;
+import org.apache.james.utils.DataProbeImpl;
+import org.apache.james.utils.SMTPMessageSender;
+import org.apache.james.utils.SpoolerProbe;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.jupiter.api.Disabled;
+import org.junit.rules.TemporaryFolder;
+
+import com.github.fge.lambdas.Throwing;
+import com.google.common.base.Stopwatch;
+
+/**
+ * This benches the efficiency at which James mailetcontainer splits emails.
+ */
+@Disabled
+public class MimeMessageBenchTest {
+    private static String CONTENT = "0123456789\r\n".repeat(1024 * 10); // 120KB message
+
+    @Rule
+    public TemporaryFolder temporaryFolder = new TemporaryFolder();
+    @Rule
+    public SMTPMessageSender messageSender = new SMTPMessageSender(DEFAULT_DOMAIN);
+
+    private TemporaryJamesServer jamesServer;
+
+    @Before
+    public void setup() throws Exception {
+        jamesServer = TemporaryJamesServer.builder()
+            .withBase(MemoryJamesServerMain.SMTP_AND_IMAP_MODULE)
+            .withMailetContainer(
+                generateMailetContainerConfiguration())
+            .build(temporaryFolder.newFolder());
+        jamesServer.start();
+
+        DataProbe dataProbe = jamesServer.getProbe(DataProbeImpl.class);
+        dataProbe.addDomain(DEFAULT_DOMAIN);
+        dataProbe.addUser("rcpt1@" + DEFAULT_DOMAIN, PASSWORD);
+        dataProbe.addUser("rcpt2@" + DEFAULT_DOMAIN, PASSWORD);
+        dataProbe.addUser("rcpt3@" + DEFAULT_DOMAIN, PASSWORD);
+        dataProbe.addUser("rcpt4@" + DEFAULT_DOMAIN, PASSWORD);
+        dataProbe.addUser("rcpt5@" + DEFAULT_DOMAIN, PASSWORD);
+    }
+
+    @After
+    public void tearDown() {
+        jamesServer.shutdown();
+    }
+
+    @Test
+    public void receivedMessagesShouldContainDeliveredToHeaders() throws Exception {
+        messageSender.connect(LOCALHOST_IP, jamesServer.getProbe(SmtpGuiceProbe.class).getSmtpPort());
+
+        Stopwatch stopwatch = Stopwatch.createStarted();
```

Comment:
I think it would be worth it to investigate jmh microbenchmarks in the future. Some useful resources: 
- https://github.com/openjdk/jmh
- https://stackoverflow.com/questions/30485856/how-to-run-jmh-from-inside-junit-tests

Reply:
Thanks! Junit integration sounds really easy to use.